app/src/main/kotlin/at/bitfire/davdroid/ui/intro/BatteryOptimizationsFragment.kt

```diff
@@ -83,6 +131,164 @@ class BatteryOptimizationsFragment: Fragment() {
+        shouldBeWhitelisted: Boolean,
+        onChangeShouldBeWhitelisted: (Boolean) -> Unit,
+        dontShowAutostart: Boolean,
+        onChangeDontShowAutostart: (Boolean) -> Unit,
+        manufacturerWarning: Boolean
+    ) {
+        val uriHandler = LocalUriHandler.current
+
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .verticalScroll(rememberScrollState())
+                .padding(8.dp)
+        ) {
+            Card {
+                Column(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(16.dp)
+                ) {
+                    Row(
+                        modifier = Modifier.fillMaxWidth(),
+                        verticalAlignment = Alignment.CenterVertically
+                    ) {
+                        Text(
+                            text = stringResource(R.string.intro_battery_title),
+                            style = MaterialTheme.typography.h6,
+                            modifier = Modifier.weight(1f)
+                        )
+                        Switch(
+                            checked = shouldBeWhitelisted,
+                            onCheckedChange = {
+                                // Only accept click events if not whitelisted
+                                if (!isWhitelisted) {
+                                    onChangeShouldBeWhitelisted(it)
+                                }
+                            },
+                            enabled = !dontShowBattery
+                        )
+                    }
+                    Text(
+                        text = stringResource(
+                            R.string.intro_battery_text,
+                            getString(R.string.app_name)
+                        ),
+                        style = MaterialTheme.typography.body1,
+                        modifier = Modifier.padding(top = 12.dp)
+                    )
+                    AnimatedVisibility(visible = !isWhitelisted) {
+                        Row(
+                            verticalAlignment = Alignment.CenterVertically
+                        ) {
+                            Checkbox(
+                                checked = dontShowBattery,
+                                onCheckedChange = onChangeDontShowBattery,
+                                enabled = !isWhitelisted
+                            )
+                            Text(
+                                text = stringResource(R.string.intro_battery_dont_show),
+                                style = MaterialTheme.typography.caption,
+                                modifier = Modifier
+                                    .clickable { onChangeDontShowBattery(!dontShowBattery) }
+                            )
+                        }
+                    }
+                }
+            }
+            AnimatedVisibility(visible = manufacturerWarning) {
```

Comment:
`manufacturerWarning` can't change except when it's loaded, so the animation just happens when the Fragment is loaded, which I find confusing.